zingstats/changes.py

```diff
@@ -0,0 +1,287 @@
+            self.merged_dt = datetime.strptime(self._merged_ts,
+                                               GerritChange.GERRIT_FORMAT)
+        self.url = '%s/changes/%s' % (parent_url, self.long_id)
+        self.review_url = '%s/%s' % (parent_url, self.number)
+
+        for revision_id in change['revisions']:
+            revision = GerritRevision(revision_id,
+                                      change['revisions'][revision_id],
+                                      '%s/revisions/%s' % (
+                                          self.url, revision_id),
+                                      self.verify_https_requests)
+            self._revisions[int(revision.number)] = revision
+
+        for message_json in change['messages']:
+            message = GerritMessage(message_json['id'],
+                                    message_json['date'],
+                                    message_json['message'])
+            self._revisions[int(message_json['_revision_number'])].add_message(
+                message)
+
+
+class Changes(object):
+    def __init__(self, url, query, projects, branches, start_dt, finish_dt,
+                 verify_https_requests=False):
+        self.url = url
+        self.query = query
+        self.projects = projects
+        self.branches = branches
+        self.start_dt = start_dt
+        self.finish_dt = finish_dt
+        self.query_start = 0
+        self.verify_https_requests = verify_https_requests
+        self.changes = dict()
+
+    def __iter__(self):
+        for long_id in self.changes:
+            yield self.changes[long_id]
+
+    def __len__(self):
+        return len(self.changes)
+
+    def add(self, change):
+        self.changes[change.long_id] = change
+
+    # TODO implement better__getitem__()
+    def __getitem__(self, key):
+        return self.changes[key]
+
+    @staticmethod
+    def pretty_json(json_data):
+        """Pretty print JSON data, usually for logging purposes."""
+        return json.dumps(json_data, sort_keys=True, indent=4,
+                          separators=(',', ': '))
+
+
+class GerritChanges(Changes):
+    def __init__(self, url, query, projects, branches, start_dt, finish_dt,
+                 verify_https_requests=False,
+                 query_size=100, max_changes=None):
+        super(GerritChanges, self).__init__(url, query, projects, branches,
+                                            start_dt, finish_dt,
+                                            verify_https_requests)
+        self.query_size = query_size
+        self.max_changes = max_changes
+
+    def gather(self):
+        log.info('Gathering gerrit changes from %s for projects: %s and branches: %s',
+                self.url,
+                ', '.join(sorted(self.projects)),
+                ', '.join(sorted(self.branches)))
+
+        results = [{'_more_changes': True}]
+        session = requests.Session()
+        while '_more_changes' in results[-1].keys() and \
```

Comment:
Using dict.get() would be clearer IMO:

`while results[-1].get('_more_changes'):`

Reply:
Definitely! :)